distribution/xfer/download_test.go

```diff
@@ -407,12 +407,10 @@ func TestMaxDownloadAttempts(t *testing.T) {
 				})
 
 			progressChan := make(chan progress.Progress)
-			progressDone := make(chan struct{})
```

Comment:
I think the intention of this was to wait for the goroutine to finish in the main test, so the wait is just missed.
Really I guess we could just pull from `progressChan` on a timeout to ensure it's closed (at the end of the test).

Reply:
1. I agree with your idea. In my new commit, I add a timeout that will call `t.Fatal()` after 5 seconds. This test normally runs 0.25s on a MacBook.
2. I think this test may also forgot to close `progressChan`. I manually checked some channels similar to `progressChan` outside test functions [under this issue](https://github.com/moby/moby/issues/37735#issuecomment-586637603). They all have a `close()`. Without close, `for range progressChan` will never return. So I add a `defer close(progressChan)`